core/templates/components/summary-tile/topic-summary-tile.component.ts

```diff
@@ -46,6 +50,11 @@ export class TopicSummaryTileComponent {
           AppConstants.ENTITY_TYPE.TOPIC, this.topicSummary.getId(),
           this.topicSummary.getThumbnailFilename());
     }
+    this.topicNameTranslationKey = this.i18nLanguageCodeService.
+      getTopicTranslationKey(
+        this.topicSummary.getId(), TranslationKeyType.TITLE);
+    this.topicNameTranslationKeyIsToBeDisplayed = this.i18nLanguageCodeService.
+      isTranslationKeyToBeDisplayed(this.topicNameTranslationKey);
```

Comment:
It seems odd for the language code service to know what is displayed and what isn't. That's more of a component thing.

Perhaps you want something like isHackyTranslationAvailable()?

Reply:
Yeah, removed it from service.